2023-11-29-diving-into-jtag-part-2.md

```diff
@@ -0,0 +1,218 @@
+---
+
+title: Diving into JTAG protocol. Part 2 — Debugging.
+description:
+    "This article explores debugging microcontrollers using JTAG, with a special focus on the ARM Cortex-M architecture and the STM32F407VG microcontroller."
+author: Aliaksandr Kavalchuk
+tags: arm, mcu, debugging, debugger
+---
+
+<!-- excerpt start -->
+
+Part 2 of a JTAG deep-dive series gives an in-depth look at interacting with a microcontroller's memory and engaging with the processor core and debug registers.
+
+<!-- excerpt end -->
+
+{% include newsletter.html %}
+
+{% include toc.html %}
+
+![image](https://github.com/memfault/interrupt/assets/33341738/8993d42a-f524-48b5-97bc-83f6c6f89db3)
+
+As noted in my previous article [Diving into JTAG protocol. Part 1 — Overview](https://medium.com/@aliaksandr.kavalchuk/diving-into-jtag-protocol-part-1-overview-fbdc428d3a16), JTAG was initially developed for testing integrated circuits and printed circuit boards. However, its potential for debugging was realized over time, and now JTAG has become the standard protocol for microcontroller debugging. Many Firmware and Embedded engineers first encountered it in this particular context.
+
+However, there’s a catch: while the use of JTAG in testing is fairly standardized when it comes to debugging, each processor architecture has its unique nuances. With that in mind, this article will focus on debugging using JTAG on the ARM Cortex-M architecture, specifically with the STM32F407VG microcontroller.
+
+For successful debugging, it’s not enough to simply know the basics of JTAG. It’s also essential to understand the workings of the ARM Debug Access Port, a key component that allows access to the controller’s internal registers and memory. I’ve detailed this module in the article [A Deep Dive into the ARM Debug Access Port](https://piolabs.com/blog/engineering/diving-into-arm-debug-access-port.html). With that groundwork laid, let’s dive into the world of debugging through JTAG.
```

Comment:
> Second link starting at beginning is broken - https://piolabs.com/blog/engineering/diving-into-arm-debug-access-port.html

This link works for me 🤔 can you try it again to be sure?

Reply:
You're right, not seeing that either - must have been user error. Thanks!